Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -593,6 +686,57 @@ struct ProjectionParameterTypes {
     }
 };
 
+struct SubtypeParameterTypes {
+    TypeIndex superType;
+    TypeIndex underlyingType;
+
+    static unsigned hash(const SubtypeParameterTypes& params)
+    {
+        return computeSubtypeHash(params.superType, params.underlyingType);
+    }
+
+    static bool equal(const TypeHash& sig, const SubtypeParameterTypes& params)
+    {
+        if (!sig.key->is<Subtype>())
+            return false;
+
+        const Subtype* subtype = sig.key->as<Subtype>();
+        if (subtype->superType() != params.superType)
+            return false;
+
+        if (subtype->underlyingType() != params.underlyingType)
+            return false;
+
+        return true;
+    }
+
+    static void translate(TypeHash& entry, const SubtypeParameterTypes& params, unsigned)
+    {
+        uint32_t displaySize;
+        const TypeDefinition& parent = TypeInformation::get(params.superType);
+        if (parent.is<Subtype>())
+            displaySize = parent.as<Subtype>()->displaySize() + 1;
+        else
+            displaySize = 1;
+
+        RefPtr<TypeDefinition> signature = TypeDefinition::tryCreateSubtype(displaySize);
+        RELEASE_ASSERT(signature);
+
+        Subtype* subtype = signature->as<Subtype>();
+        subtype->getSuperType() = params.superType;
```

Comment:
If this is an accessor, I think we should drop the get prefix

Reply:
I agree that the naming is a bit strange, but that's the convention that the Wasm TypeDefinition code currently follows. Here's an example for functions:

  * `getReturnType` (accessor): https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/WasmTypeDefinition.h#183
  * `returnType` (getter): https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/WasmTypeDefinition.h#157

so I made `Subtype` consistent with that.